src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -2056,13 +2221,28 @@ void MFEMSidreDataCollection::reconstructFields()
     // Filter out the non-user-registered attribute fields
     if(!field_grp->hasView("association"))
     {
-      // FiniteElementCollection
-      auto basis_name = field_grp->getView("basis")->getString();
-      m_fecolls.emplace_back(mfem::FiniteElementCollection::New(basis_name));
+      int vdim = 1;  // The default
+      // GridFunction vs QuadratureFunction
+      bool is_gridfunc = true;
+      // FiniteElementCollection/QuadratureSpace construction
+      const std::string basis_name = field_grp->getView("basis")->getString();
+      // Check if it's a QuadratureFunction
+      if(basis_name.find("QF") == 0)
+      {
+        // The vdim is being overwritten here with the value in the basis string so we
+        // can correctly construct the QuadratureFunction
+        m_quadspaces.emplace_back(
```

Comment:
Potential future optimization: Should `m_quadspaces` be a map from string name to `QuadratureSpace`? 
(This would need to be unique for each mesh, but `MFEMSidreDataCollection` is only over a single mesh)

Reply:
m_quadspaces (like its counterpart m_fespaces) is just used for managing ownership - its members are never accessed directly

Reply:
I was thinking more in terms of memory savings.
(I haven't looked into the storage costs for `QuadratureSpace`, but `FiniteElementSpace` contains several indexing several arrays)

Reply:
@kennyweiss I was thinking of doing something along the lines of what you were talking about for FES and QS in a follow-up PR that would be more memory optimization focused. I know for my ECP application that'll be a useful feature to have, since all my data shares the same QS and most of the FES are also shared as well.

Reply:
Created an issue for this - #533